pkg/backend/apply.go

```diff
@@ -125,23 +125,15 @@ func PreviewThenPrompt(ctx context.Context, kind apitype.UpdateKind, stack Stack
 		return plan, changes, nil
 	}
 
+	infoPrefix := "\b" + op.Opts.Display.Color.Colorize(colors.SpecWarning+"info: "+colors.Reset)
 	if kind != apitype.UpdateUpdate {
-		// If not an update, we can continue
+		// If not an update, we can skip displaying warnings
 	} else if countResources(events) == 0 {
 		// This is an update and there are no resources being CREATED
-		emptyStackWarningMessage := "\b" + op.Opts.Display.Color.Colorize(
-			fmt.Sprintf("%sinfo:%s There are no resources found in this update.\n",
-				colors.SpecInfo,
-				colors.Reset))
-		fmt.Printf(emptyStackWarningMessage)
+		fmt.Print(infoPrefix, "There are no resources in your stack(other than the stack resource).\n\n")
 	} else if !hasChanges(events) {
 		// This is an update and there are no resources being UPDATED
-		noChangesWarningMessage := "\b" + op.Opts.Display.Color.Colorize(
-			fmt.Sprintf("%sinfo:%s There are no changes found in this update.\n",
-				colors.SpecInfo,
-				colors.Reset))
-		fmt.Printf(noChangesWarningMessage)
-
+		fmt.Print(infoPrefix, "This update does not contain any changes to resources.\n\n")
```

Comment:
Does this read better than the previous

> info: There are no changes found in this update.

Reply:
I think it does read better.

It still does feel repetitive to me. Feel free to ignore me, but reading the goals in the issue

> more than likely, folks don't mean to do this, but a lot of new users may do this by accident; if we can message them ahead of time, it may help them better understand what is happening and adjust ahead of running the update

> can we message them that this was probably not attended in clear language, along with some suggestions for what they should do next, e.g., save. happy to work with someone to figure out the details of how we want to solve the issue.

I don't feel like we addressed them? Should the message focus on a CTA?

Reply:
Apologies if I'm reading this wrong, but I wouldn't consider "no changes" something we should warn users about or even message as something out of the ordinary. Quite often "no changes" is just what you want to see.

Reply:
> Apologies if I'm reading this wrong, but I wouldn't consider "no changes" something we should warn users about or even message as something out of the ordinary. Quite often "no changes" is just what you want to see.

I think the message is to provide confirmation to novice users that the diff summary means what they think it might mean.

I marked it as `info`, but used the warn coloring to indicate where the error message starts and to draw attention and the purple doesn't show up well on a black terminal.

The reason I thought to even use the warn coloring was from:
```
warning: A new version of Pulumi is available. To upgrade from version '3.35.3' to '3.39.0', run 
   $ curl -sSL https://get.pulumi.com | sh
```

but I never realized that it actually said `warning`. should I open an issue to change that to an info message and we can use the info coloring in this location and then we can start to use the info coloring more?